acme/acme/crypto_util.py

```diff
@@ -136,22 +136,16 @@ def probe_sni(name, host, port=443, timeout=300,
 
     socket_kwargs = {'source_address': source_address}
 
-    host_protocol_agnostic = host
-    if host == '::' or host == '0':
-        # https://github.com/python/typeshed/pull/2136
-        # while PR is not merged, we need to ignore
-        host_protocol_agnostic = None
```

Comment:
Can you explain to me why this code is no longer needed? It's quite possible it's not, but it's to immediately clear to me that is the case.

Reply:
In fact, the PR reference is for something else. The fact that host could by none in the mypi type support. I do not know why there where this code. Is it to use ipv6 localhost IP in a code designed for ipv4 ? The name `host_protocol_agnostic` seems to indicate that.

But empirically, its seems that this is not needed. All tests are working without it. Better, more tests on Windows are working untouched without it.

Reply:
Introduced by 239184882e1694bee62b12e43e3d4e0a1f08b9da to add IPV6 support. I will check the impacts.

Reply:
Ok here the thing. Calling socket.get_connection() will pass host and port to socket.getaddrinfo(), that wraps call for the socket.getaddrinfo() method in the C API . This method returns the IP addresses that make sense for given host to be binded to.

It is platform dependant, but here are general rules:
`''` will map to any external address available in IPV4 and IPV6 for the host: it is basically 'open a socket available to the world'
`':'` will map to the 'unspecified' IPV6 address `::0`, that corresponds to any external address available in IPV6
`0` will map to the 'unspecified' IPV6 address `0.0.0.0`, that corresponds to any external address available in IPV4
`None` will map to both `::1` and `127.0.0.1`, so the loopback addresses for IPV4 and IPV4.

So anyway converting `::` and `0` to `None` changes the meaning of the binding. For the first, it narrows the scope to loopback only, for the second it opens the scope to both IPV4 and IPV6 loopbacks.

I do not think this is a good idea, and the code should be removed.

Reply:
Thanks for doing that digging. It makes sense to me and I agree it can be removed.